### PR TITLE
List generator

### PR DESCRIPTION
**Pull request by HamzaBenyazid**

### CODE REVIEW

File: src/main/java/io/javarig/generator/ListGenerator.java [hunk 1]
```diff
@@ -1,8 +1,33 @@
...
+    private int minSizeInclusive = 5;
+    private int maxSizeExclusive = 15;
+
+    private ParameterizedType type;
+
     @Override
-    public Object generate() {
-        return null;
+    public List<Object> generate() {
+        int randomSize = random.nextInt(getMinSizeInclusive(), getMaxSizeExclusive());
```
Comment (oubaydos) [inline src/main/java/io/javarig/generator/ListGenerator.java:21]: this fails to detect an illegalArgumentExceptin when minSize < maxSize but they are both negatives, ex: -10,-1

File: src/main/java/io/javarig/generator/ListGenerator.java [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
+        ParameterizedType parameterizedType = getType();
+        Type inputListType = parameterizedType.getActualTypeArguments()[0];
+        List<Object> outputList = new ArrayList<>(randomSize);
+        for (int i = 0; i < randomSize; i++) {
+            try {
+                outputList.add(randomGenerator.generate(Class.forName(inputListType.getTypeName())));
+            } catch (Exception ignore) {
```
Comment (oubaydos) [inline src/main/java/io/javarig/generator/ListGenerator.java:28]: todo

File: src/main/java/io/javarig/generator/LocalDateGenerator.java [hunk 3]
```diff
@@ -1,8 +1,12 @@
 package io.javarig.generator;
 
-public class LocalDateGenerator implements Generator{
+import java.time.Instant;
+
+import static java.time.ZoneOffset.UTC;
+
+public class LocalDateGenerator implements Generator {
     @Override
     public Object generate() {
```
Comment (oubaydos) [inline src/main/java/io/javarig/generator/LocalDateGenerator.java:9]: change returntype

Comment (oubaydos) [inline src/main/java/io/javarig/generator/LocalDateGenerator.java]: change returntype

**Review by oubaydos — CHANGES_REQUESTED** [review]

File: src/main/java/io/javarig/generator/ListGenerator.java (hunk 1 above)
Comment (HamzaBenyazid) [reply src/main/java/io/javarig/generator/ListGenerator.java:21]: Why not just let the exception to be thrown ? the minSize o maxSize shouldn't be negative anyway

File: src/main/java/io/javarig/generator/LocalDateGenerator.java (hunk 3 above)
Comment (HamzaBenyazid) [reply src/main/java/io/javarig/generator/LocalDateGenerator.java]: done

**Review by oubaydos — APPROVED** [review]

**Pull request merged**